cmd/xmtpd-cli/commands/version.go

```diff
@@ -0,0 +1,178 @@
+package commands
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/spf13/cobra"
+	"github.com/xmtp/xmtpd/cmd/xmtpd-cli/options"
+	"go.uber.org/zap"
+)
+
+func versionCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:          "version",
+		Short:        "Get version of contract",
+		SilenceUsage: true,
+	}
+	cmd.AddCommand(
+		getVersionCmd(),
+	)
+	return cmd
+}
+
+// --- pause ---
+func getVersionCmd() *cobra.Command {
+	var target options.Target
+	cmd := &cobra.Command{
+		Use:          "get",
+		Short:        "todo",
+		SilenceUsage: true,
+		RunE: func(*cobra.Command, []string) error {
+			return getVersionHandler(target)
+		},
+	}
+	cmd.Flags().
+		Var(&target, "target", "settlement-chain-gateway|payer-registry|distribution-manager")
+	_ = cmd.MarkFlagRequired("target")
```

Comment:
`version()` is owned from the IVersionable interface. We should be able to extend `dev/gen/abi` to generate the go bindings for that interface and call version from any arbitrary contract, and would make `getVersionHandler` way easier, needing just a target address and a go-eth client.

Reply:
probably true. It is unlikely that I will rework this any time soon...